pkgs/desktops/gnome/extensions/hanabi/default.nix

```diff
@@ -0,0 +1,89 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  appstream-glib,
+  clapper,
+  gettext,
+  gjs,
+  glib,
+  gobject-introspection,
+  gst_all_1,
+  gtk4,
+  libadwaita,
+  makeWrapper,
+  meson,
+  ninja,
+  nodePackages,
+  shared-mime-info,
+  wrapGAppsHook4,
+}:
+
+stdenv.mkDerivation (finalAttrs: rec {
+  pname = "gnome-shell-extension-hanabi";
+  version = "0-unstable-2024-12-01";
+
+  src = fetchFromGitHub {
+    owner = "jeffshee";
+    repo = "gnome-ext-hanabi";
+    rev = "3ed1e3c4e6a533d3f8e5bf38fbf994e6d134f473";
```

Comment:
According to [README](https://github.com/jeffshee/gnome-ext-hanabi?tab=readme-ov-file#installation), GNOME 47 support is in a separate branch. If a non-main branch is used, it should be mentioned in a comment next to `rev`.